index.d.ts

```diff
@@ -28,11 +30,13 @@ declare module 'apollo-datasource-mongodb' {
       | (string | number | boolean | ObjectId)[]
   }
 
+  type MongooseDocumentOrMongoCollection<T> = MongoCollection<T> | MongooseDocument
+
   export interface Options {
     ttl: number
   }
 
-  export class MongoDataSource<TData, TContext = any> extends DataSource<
+  export class MongoDataSource<TData extends MongooseDocumentOrMongoCollection<any>, TContext = any> extends DataSource<
```

Comment:
I will drop this part of the code, it is an unnecessary type constraint. I think it was added because of the way we used it internally, but for general usage, it isn't necessary.